test/functional/autocmd/signal_spec.lua

```diff
@@ -8,39 +8,88 @@ local fn = n.fn
 local next_msg = n.next_msg
 local is_os = t.is_os
 local skip = t.skip
+local read_file = t.read_file
+local feed = n.feed
+local retry = t.retry
 
 if skip(is_os('win'), 'Only applies to POSIX systems') then
   return
 end
 
-local function posix_kill(signame, pid)
-  os.execute('kill -s ' .. signame .. ' -- ' .. pid .. ' >/dev/null')
-end
+describe("'autowriteall' on signal exit", function()
+  before_each(clear)
+
+  local function test_deadly_sig(signame, awa, should_write)
+    local testfile = 'Xtest_SIG' .. signame .. (awa and '_awa' or '_noawa')
+    local teststr = 'Testaaaaaaa'
+
+    if awa then
+      command('set awa')
+    end
+
+    command('edit ' .. testfile)
+    feed('i' .. teststr)
+    vim.uv.kill(fn.getpid(), signame)
+
+    retry(nil, 5000, function()
+      eq((should_write and (teststr .. '\n') or nil), read_file(testfile))
+    end)
+  end
+
+  it('dont write if SIGTERM & awa on', function()
+    test_deadly_sig('sigterm', true, false)
+  end)
+  it('dont write if SIGTERM & awa off', function()
+    test_deadly_sig('sigterm', false, false)
+  end)
```

Comment:
can we combine each pair of tests like this:
```suggestion
  it('SIGTERM', function()
    test_deadly_sig('sigterm', true, false)
    test_deadly_sig('sigterm', false, false)
  end)
```

Reply:
Since test_deadly_sig() terminate the neovim instance and contain an eq(), I'm not sure we can easly combine them